src/core/ad.jl

```diff
@@ -66,26 +66,23 @@ function gradient_logp(
     model::Model,
     sampler::Sampler
 )
-    ad_type = getADtype(sampler)
-    if ad_type <: ForwardDiffAD
-        return gradient_logp_forward(θ, vi, model, sampler)
-    else ad_type <: TrackerAD
-        return gradient_logp_reverse(θ, vi, model, sampler)
-    end
+    return gradient_logp(getADbackend(sampler), θ, vi, model, sampler)
 end
 
 """
-gradient_logp_forward(
+gradient_logp(
```

Comment:
Same here, the function signature should be indented:
```suggestion
    gradient_logp(
```
(I guess you want to adjust the following lines as well).
BTW maybe one could make use of DocStringExtensions here.

Reply:
Maybe in another PR.